docs/installation.md

```diff
@@ -10,74 +10,53 @@ These steps are required once per machine.
-data are produced by an sbt plugin. This plugin is currently not published, so
-you will need to copy paste it on your machine.
+data are produced by an sbt plugin.
 
-Here's the source of the plugin:
-https://github.com/scalameta/metals/blob/master/project/MetalsPlugin.scala
+You can install the plugin with
 
-Copy the source to either (depending on your sbt version):
-
-* (sbt 0.13) `~/.sbt/0.13/plugins/MetalsPlugin.scala`
-* (sbt 1.+) `~/.sbt/1.0/plugins/MetalsPlugin.scala`
-
-## Step 2 - build the VSCode extension
-
-The VSCode extension is not yet published on the Marketplace, so you'll need to
-build it locally.
-
-* Make sure you have installed `node`, `npm` and VS Code.
-* `cd vscode-extension`
-* `npm install`
-* `npm run build`
-* `code --install-extension metals-0.1.0.vsix`
-
-## Step 3 - publish the server locally
+```
+addSbtPlugin("org.scalameta" % "sbt-metals" % "<version>")
+```
 
-From the repo root run `sbt publishLocal`
+## Step 2 - VSCode extension
+The VSCode extension is published on the Marketplace. You can open VSCode and search for it.
 
 # Per-project setup
 
 These steps are required on each project.
 
-## Step 1 - add semanticdb-scalac compiler plugin to your project
+## Quick-start
+The quickest way to get started with Metals is to use the `metalsSetup` command in sbt.
+
+```
+sbt
+> metalsSetup
+```
 
-Some features like definition/references/hover rely on artifacts produced by a
-compiler plugin called `semanticdb-scalac`. There are two alternative ways to
-install `semanticdb-scalac`.
+The command will create the necessary metadata in the `.metals` directory
+(which you should not checkout into version control) and setup the `semanticdb-scalac` compiler
+plugin for the current sbt ession.
 
-The first option is to enable `semanticdb-scalac` permanently for your project
-in `build.sbt` with:
+Note that you will need to invoke `metalsSetup` (or `semanticdbEnable`) whenevery you close and
+re-open sbt. For a more persistent setup, keep reading.
 
-```scala
- addCompilerPlugin("org.scalameta" % "semanticdb-scalac" % "2.1.8" cross CrossVersion.full)
-scalacOptions += "-Yrangepos"
-```
+## Persisting the semanticdb-scalac compiler plugin
+Some features like definition/references/hover rely on artifacts produced by a compiler plugin
+called `semanticdb-scalac`.
 
-The second option is to enable `semanticdb-scalac` only for an active sbt
-session by running `semanticdbEnable` from the sbt shell.
+`metalsSetup` enables the plugin on the current session (by invoking `semanticdbEnable`), but you
+can choose to enable it permanently on your project by adding these two settings in your sbt build
+definition:
 
 ```scala
-$ sbt
-> semanticdbEnable // automatically runs libraryDependencies += compilerPlugin(...)
-> compile // re-compile project with semanticdb-scalac compiler plugin
-> ...
+addCompilerPlugin("org.scalameta" % "semanticdb-scalac" % "2.1.8" cross CrossVersion.full)
```

Comment:
sbt-metals exposes `semanticdbSettings`, I'm wondering if we should encourage using that instead?

I'm a bit torn which approach is the best, on one hand I think it's good that users are aware what scalameta version they're on but on the other hand it's more to document and users need to update the dependency with new releases.

Reply:
sbt-scalafix v0.6 will no longer automatically enable semanticdb-scalac because I've found automatic installation is fragile. `semanticdbSettings` only works if it appears after `libraryDependencies +=` and `scalacOptions +=`

One alternative is to expose 

```scala
val semanticdbScalac = "org.scalameta" % "semanticdb-scalac" % "2.1.8" cross CrossVersion.full
```

and tell users to 
```
addCompilerPlugin(semanticdbScalac)
```

This has the benefit of being clear to the user while avoiding the need for users to upgrade scalameta.

Reply:
How about using sbt's [`onLoad`](https://www.scala-sbt.org/1.0/docs/Howto-Startup.html)? Something like

```scala
onLoad in Global ~= {
  { st: State => semanticdbEnableCommand.key.label :: st } compose _
}
```

(just an idea)

Reply:
I like the 

```

Reply:
Is `addCompilerPlugin(semanticdbScalac)` enough on its own? or is `semanticdbSettings` still needed in addition to it?